6_nodejs/week2/app.js

```diff
@@ -0,0 +1,87 @@
+import express from "express";
+// import {bodyParser} from 'body-parser'
+import fs from "fs";
+
+const port = process.env.PORT || 3000;
+const data = JSON.parse(fs.readFileSync("./document.json"));
+const app = express();
+
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+// GET /search
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+  if (query) {
+    const filteredData = data.filter((item) => {
+      return Object.values(item).some(
+        (value) => typeof value === "string" && value.includes(query)
+      );
+    });
+    if (filteredData.length === 0) {
+      return res.send("not found");
+    }
+    return res.json({ data: filteredData, message: "ok" });
+  }
+  res.json({ data: data, message: "ok" });
+});
+
+// GET /documents/:id
+app.get("/search/:id", (req, res) => {
+  const dataId = +req.params.id;
+  const myData = data.find((data) => data.id === dataId);
+  if (!myData) {
+    return res.status(404).json({ data: null, message: "not found" });
+  }
+  res.json({ data: myData, message: "ok" });
```

Comment:
This is right and clean but if you want to follow one response at end rule it could be this way:

```
const result = data.find((data) => data.id === +req.params.id)
  const response = result.length ? { status: 200, message: 'ok' } : { status: 404, message: 'not found' }
  
  res.status(response.status).json({ data: response.message, message: response.message })
```